UefiCpuPkg/PiSmmCpuDxeSmm/PiSmmCpuDxeSmm.c

```diff
@@ -391,3 +391,43 @@ PiCpuSmmEntry (
 
   return Status;
 }
+
+/**
+  Initialize SMM profile in SmmReadyToLock protocol callback function.
+
+  @param  Protocol   Points to the protocol's unique identifier.
+  @param  Interface  Points to the interface instance.
+  @param  Handle     The handle on which the interface was installed.
+
+  @retval EFI_SUCCESS SmmReadyToLock protocol callback runs successfully.
+**/
+EFI_STATUS
+EFIAPI
+InitSmmProfileCallBack (
+  IN CONST EFI_GUID  *Protocol,
+  IN VOID            *Interface,
+  IN EFI_HANDLE      Handle
+  )
+{
+  EFI_STATUS                 Status;
+  EFI_SMM_VARIABLE_PROTOCOL  *SmmProfileVariable;
+
+  //
+  // Locate SmmVariableProtocol.
+  //
+  Status = gMmst->MmLocateProtocol (&gEfiSmmVariableProtocolGuid, NULL, (VOID **)&SmmProfileVariable);
+  ASSERT_EFI_ERROR (Status);
+
+  //
+  // Save to variable so that SMM profile data can be found.
+  //
+  SmmProfileVariable->SmmSetVariable (
+                        SMM_PROFILE_NAME,
+                        &gEfiCallerIdGuid,
+                        EFI_VARIABLE_BOOTSERVICE_ACCESS | EFI_VARIABLE_RUNTIME_ACCESS,
+                        sizeof (mSmmProfileBase),
+                        &mSmmProfileBase
+                        );

```

Comment:
Should also check the status of SmmSetVariable.

Reply:
Thanks, will update in coming patch.